lib/web_ui/lib/src/engine/html/render_vertices.dart

```diff
@@ -195,6 +198,31 @@ class _WebGlRenderer implements _GlRenderer {
   /// Browsers that support OffscreenCanvas and the transferToImageBitmap api
   /// will return ImageBitmap, otherwise will return CanvasElement.
   Object? drawRect(ui.Rect targetRect, _GlContext gl, _GlProgram glProgram,
+      NormalizedGradient gradient, int widthInPixels, int heightInPixels, ) {
+    drawRectToGl(targetRect, gl, glProgram, gradient, widthInPixels, heightInPixels);
+    Object? image = gl.readPatternData();
+    gl.bindArrayBuffer(null);
+    gl.bindElementArrayBuffer(null);
+    return image;
+  }
+
+  /// Renders a rectangle using given program into an image resource.
+  ///
+  /// Browsers that support OffscreenCanvas and the transferToImageBitmap api
+  /// will return ImageBitmap, otherwise will return CanvasElement.
+  String drawRectToImageUrl(ui.Rect targetRect, _GlContext gl, _GlProgram glProgram,
+      NormalizedGradient gradient, int widthInPixels, int heightInPixels, ) {
+    drawRectToGl(targetRect, gl, glProgram, gradient, widthInPixels, heightInPixels);
+    final String imageUrl = gl.toImageUrl();
+    gl.bindArrayBuffer(null);
+    gl.bindElementArrayBuffer(null);
+    return imageUrl;
+  }
+
+  /// Renders a rectangle using given program into an image resource.
```

Comment:
It's unclear what image resource this refers to and how that image resource can be used after calling this method.

Reply:
Done.